lib/puppet/application/resource.rb

```diff
@@ -15,6 +15,7 @@ def preinit
   option("--verbose", "-v")
   option("--edit", "-e")
   option("--to_yaml", "-y")
+  option('--fail-when-not-found', '-f')
```

Comment:
@cthorn42 sorry just noticed this. I think originally we were thinking this was specific to absent services, but really it's any error that occurs when trying to modify the system state (because of the call to `Puppet::Resource.indirection.save`). So maybe just `--fail` like curl?

Also in the future I could see us using the same command line option for other applications when we should be returning a non-zero exit code due to a failure but don't currently.

Reply:
👍 I like that, a lot more succinct.